engine/api/migrate/hatchery_cmd.go

```diff
@@ -0,0 +1,184 @@
+package migrate
+
+import (
+	"encoding/base64"
+	"encoding/json"
+
+	"github.com/go-gorp/gorp"
+
+	"github.com/ovh/cds/engine/api/cache"
+	"github.com/ovh/cds/engine/api/worker"
+	"github.com/ovh/cds/sdk"
+	"github.com/ovh/cds/sdk/log"
+)
+
+type deprecatedOpenstackModelData struct {
+	Image    string `json:"os,omitempty"`
+	Flavor   string `json:"flavor,omitempty"`
+	UserData string `json:"user_data,omitempty"`
+}
+
+type deprecatedVSphereModelData struct {
+	OS       string `json:"os"`
+	UserData string `json:"user_data"` //Commands to execute when create vm model
+}
+
+// HatcheryCmdMigration useful to set default tags to git.branch git.author
+func HatcheryCmdMigration(store cache.Store, DBFunc func() *gorp.DbMap) {
+	db := DBFunc()
+
+	log.Info("HatcheryCmdMigration> Begin")
+
+	wms, err := worker.LoadWorkerModels(db)
+	if err != nil {
+		log.Warning("HatcheryCmdMigration> Cannot load worker models : %v", err)
+		return
+	}
+
+	for _, wmTmp := range wms {
+		tx, errTx := db.Begin()
```

Comment:
The migrate script have to migrate only if Shell is empty, no? To avoid override user values at each restart of api.